core/src/main/scala/akka/persistence/cassandra/journal/TagWriters.scala

```diff
@@ -138,19 +152,23 @@ import akka.util.ByteString
     case WriteTagScanningTick =>
       writeTagScanning()
 
-    case PidRecovering(pid, tagProgresses: Map[Tag, TagProgress]) =>
-      val replyTo = sender()
+    case PersistentActorStarting(pid, tagProgresses: Map[Tag, TagProgress], persistentActor) =>
       val missingProgress = tagActors.keySet -- tagProgresses.keySet
+      log.debug("Persistent actor [{}] with pid [{}] starting with progress [{}]. Tags to reset as not in progress: [{}]", persistentActor, pid, tagProgresses, missingProgress)
+
+      // EventsByTagMigration uses dead letters are there are no real actors
+      if (persistentActor != context.system.deadLetters) {
+        currentPersistentActors += (pid -> persistentActor)
+        context.watchWith(persistentActor, PersistentActorTerminated(pid, persistentActor))
```

Comment:
No problems with watching each local persistent actor, at scale?

Reply:
I think we should be able to handle that. A potential problem could be a storm of terminations when shutting down the ActorSystem.

Reply:
should we put this feature behind a flag so that doesn't affect users that don't use a large number of short lived actors?

Reply:
We can always add that later